sw/device/lib/dif/dif_csrng_unittest.cc

```diff
@@ -265,5 +265,65 @@ TEST_F(GenerateEndTest, ReadBadArgs) {
   EXPECT_EQ(dif_csrng_generate_end(nullptr, &data, /*len=*/1), kDifCsrngBadArg);
 }
 
+class GetInternalStateTest : public DifCsrngTest {};
+
+TEST_F(GetInternalStateTest, GetInternalStateOk) {
+  dif_csrng_internal_state_id_t instance_id = kCsrngInternalStateIdSw;
+
+  EXPECT_WRITE32(CSRNG_INT_STATE_NUM_REG_OFFSET,
+                 {
+                     {CSRNG_INT_STATE_NUM_INT_STATE_NUM_OFFSET,
+                      static_cast<uint32_t>(instance_id)},
+                 });
+  EXPECT_WRITE32(CSRNG_HALT_MAIN_SM_REG_OFFSET,
+                 {
+                     {CSRNG_HALT_MAIN_SM_HALT_MAIN_SM_BIT, true},
+                 });
+  EXPECT_READ32(CSRNG_MAIN_SM_STS_REG_OFFSET,
+                {
+                    {CSRNG_MAIN_SM_STS_MAIN_SM_STS_BIT, true},
+                });
+
+  dif_csrng_internal_state_t expected = {
+      .reseed_counter = 1,
+      .v = {1, 2, 3, 4},
+      .key = {1, 2, 3, 4, 5, 6, 7, 8},
+      .instantiated = true,
+      .fips_compliance = true,
+  };
+  EXPECT_READ32(CSRNG_INT_STATE_VAL_REG_OFFSET, expected.reseed_counter);
+  for (size_t i = 0; i < 4; ++i) {
+    EXPECT_READ32(CSRNG_INT_STATE_VAL_REG_OFFSET, expected.v[i]);
+  }
+  for (size_t i = 0; i < 8; ++i) {
+    EXPECT_READ32(CSRNG_INT_STATE_VAL_REG_OFFSET, expected.key[i]);
+  }
+  EXPECT_READ32(CSRNG_INT_STATE_VAL_REG_OFFSET, 3);
+
+  EXPECT_WRITE32(CSRNG_HALT_MAIN_SM_REG_OFFSET,
+                 {
+                     {CSRNG_HALT_MAIN_SM_HALT_MAIN_SM_BIT, false},
+                 });
+  EXPECT_READ32(CSRNG_MAIN_SM_STS_REG_OFFSET,
+                {
+                    {CSRNG_MAIN_SM_STS_MAIN_SM_STS_BIT, false},
+                });
+
+  dif_csrng_internal_state_t got;
+  EXPECT_EQ(dif_csrng_get_internal_state(&csrng_, instance_id, &got),
+            kDifCsrngOk);
+}
```

Comment:
nit: should this also check that `got` is equal to `expected`?

Reply:
Done.